sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessorBase.cs

```diff
@@ -0,0 +1,729 @@
+    /// </summary>
+    ///
+    /// <typeparam name="T">An underlying type with additional information that fully describes the context in which a partition is being processed.</typeparam>
+    ///
+    public abstract class EventProcessorBase<T>
+    {
+        /// <summary>The seed to use for initializing random number generated for a given thread-specific instance.</summary>
+        private static int s_randomSeed = Environment.TickCount;
+
+        /// <summary>The random number generator to use for a specific thread.</summary>
+        private static readonly ThreadLocal<Random> RandomNumberGenerator = new ThreadLocal<Random>(() => new Random(Interlocked.Increment(ref s_randomSeed)), false);
+
+        /// <summary>The primitive for synchronizing access during start and close operations.</summary>
+        private readonly SemaphoreSlim RunningTaskSemaphore = new SemaphoreSlim(1, 1);
+
+        /// <summary>
+        ///   The fully qualified Event Hubs namespace that the processor is associated with.  This is likely
+        ///   to be similar to <c>{yournamespace}.servicebus.windows.net</c>.
+        /// </summary>
+        ///
+        public abstract string FullyQualifiedNamespace { get; }
+
+        /// <summary>
+        ///   The name of the Event Hub that the processor is connected to, specific to the
+        ///   Event Hubs namespace that contains it.
+        /// </summary>
+        ///
+        public abstract string EventHubName { get; }
+
+        /// <summary>
+        ///   The name of the consumer group this event processor is associated with.  Events will be
+        ///   read only in the context of this group.
+        /// </summary>
+        ///
+        public abstract string ConsumerGroup { get; }
+
+        /// <summary>
+        ///   A unique name used to identify this event processor.
+        /// </summary>
+        ///
+        public abstract string Identifier { get; }
+
+        /// <summary>
+        ///   The active policy which governs retry attempts for the
+        ///   processor.
+        /// </summary>
+        ///
+        protected abstract EventHubsRetryPolicy RetryPolicy { get; }
+
+        /// <summary>
+        ///   The minimum amount of time to be elapsed between two load balancing verifications.
+        /// </summary>
+        ///
+        protected virtual TimeSpan LoadBalanceUpdate => TimeSpan.FromSeconds(10);
+
+        /// <summary>
+        ///   The minimum amount of time for an ownership to be considered expired without further updates.
+        /// </summary>
+        ///
+        protected virtual TimeSpan OwnershipExpiration => TimeSpan.FromSeconds(30);
+
+        /// <summary>
+        ///   A <see cref="CancellationTokenSource"/> instance to signal the request to cancel the current running task.
+        /// </summary>
+        ///
+        private CancellationTokenSource RunningTaskTokenSource { get; set; }
```

Comment:
(extreme) nit:  Typically the members flow from public (most visible) to private (least visible) or vice-versa.   We've been doing the former to date.

Reply:
I changed so many access modifiers in this class that I completely forgot to pay attention to this. We'll be moving everything to the derived class for GA anyway, so I think there's no need to worry now.